third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv2.cpp

```diff
@@ -516,6 +516,8 @@ LogicalResult convertDot(const LLVMTypeConverter *typeConverter,
 
   const auto &mmaInstructions =
       isTuring ? mmaInstrPtxTuring : mmaInstrPtxAmpere;
+  assert(mmaInstructions.count(mmaType) &&
+    "Unsupported MMA instruction for the given mma type");
```

Comment:
What does the error look like here? If we use `op.emitError` do we get a better error message?

Reply:
Let me try. It's currently:

```
python: /home/kzhou6/Code/triton/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv2.cpp:520: llvm::LogicalResult convertDot(const mlir::LLVMTypeConverter*, mlir::ConversionPatternRewriter&, mlir::Location, mlir::Value, mlir::Value, mlir::Value, mlir::Value, mlir::Value, mlir::Value, mlir::Value, mlir::triton::DotOp, mlir::triton::DotOpAdaptor, bool): Assertion `mmaInstructions.count(mmaType) && "Unsupported MMA instruction for the given mma type"' failed.
```

Reply:
Yeah, indeed it's better

```
/home/kzhou6/test.py:19:18: error: Unsupported MMA type found
    C = tl.dot(A, B, out_dtype=tl.float16)
```

Reply:
I think maybe a lot of backend asserts can be converted to emitErrors

Reply:
Nice, that's a lot better.